### PR TITLE
feat(core-flows, types, medusa): Add Update location level endpoint for api-v2

### PR DESCRIPTION
**Pull request by pKorsholm**

### CODE REVIEW

**Review by riqwan — APPROVED** [review]
nice! 🔥

File: packages/medusa/src/api-v2/admin/inventory-items/[id]/location-levels/[location_id]/route.ts [hunk 1]
```diff
@@ -45,3 +47,37 @@ export const DELETE = async (req: MedusaRequest, res: MedusaResponse) => {
...
+      variables: {
+        inventory_item_id,
+        location_id,
+      },
+      fields: req.retrieveConfig.select as string[],
+    })
+  )
+
+  res.status(200).json({
+    inventory_level,
```
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/inventory-items/[id]/location-levels/[location_id]/route.ts]: nit: Respond with parent entity inventory item to comply with our convention. I also believe responding with the inventory_level would be a breaking change, and we might as well avoid that

**Review by olivermrbl — APPROVED** [review]
LGTM, just a minor comment!

File: integration-tests/modules/__tests__/inventory/index.spec.ts [hunk 2]
```diff
@@ -284,6 +284,97 @@ medusaIntegrationTestRunner({
...
+        beforeEach(async () => {
+          const invItemReps = await api.post(
+            `/admin/inventory-items`,
+            { sku: "test-sku" },
+            adminHeaders
+          )
+
+          inventoryItemId = invItemReps.data.inventory_item.id
+
+          const sl = await appContainer
```
Comment (adrien2p) [inline integration-tests/modules/__tests__/inventory/index.spec.ts]: todo: as per our convention (old one) we shouldn't use one or two letter variable name, only in inline function

Comment (adrien2p) [reply integration-tests/modules/__tests__/inventory/index.spec.ts]: I know I can just read the resolve to understand that it is the stock location module and therefore it is probably a stock location, but better getting use to it now to stop doing that :)

File: integration-tests/modules/__tests__/inventory/index.spec.ts [hunk 3]
```diff
@@ -284,6 +284,97 @@ medusaIntegrationTestRunner({
...
+            `/admin/inventory-items/${inventoryItemId}/location-levels/${locationId}`,
+            {
+              stocked_quantity: 15,
+              incoming_quantity: 5,
+            },
+            adminHeaders
+          )
+
+          expect(result.status).toEqual(200)
+          expect(result.data.inventory_level).toEqual(
```
Comment (adrien2p) [inline integration-tests/modules/__tests__/inventory/index.spec.ts]: todo: as per the convention I believe it should return the inventory_item, @olivermrbl is that correct?

File: integration-tests/modules/__tests__/inventory/index.spec.ts [hunk 4]
```diff
@@ -284,6 +284,97 @@ medusaIntegrationTestRunner({
...
+                incoming_quantity: 5,
+              },
+              adminHeaders
+            )
+            .catch((e) => e)
+
+          expect(error.response.status).toEqual(404)
+          expect(error.response.data).toEqual({
+            type: "not_found",
+            message: `Item does-not-exist is not stocked at location ${locationId}`,
```
Comment (adrien2p) [inline integration-tests/modules/__tests__/inventory/index.spec.ts:387]: suggestion: The message here let you think that the item exists but is not stocked there, but in fact the item does not exists, should we change the message?

File: packages/medusa/src/api-v2/admin/inventory-items/[id]/location-levels/[location_id]/route.ts [hunk 5]
```diff
@@ -45,3 +47,37 @@ export const DELETE = async (req: MedusaRequest, res: MedusaResponse) => {
...
+  }
+
+  const [inventory_level] = await remoteQuery(
+    remoteQueryObjectFromString({
+      entryPoint: "inventory_level",
+      variables: {
+        inventory_item_id,
+        location_id,
+      },
+      fields: req.retrieveConfig.select as string[],
```
Comment (adrien2p) [inline packages/medusa/src/api-v2/admin/inventory-items/[id]/location-levels/[location_id]/route.ts]: todo: req.remoteQueryConfig.fields, it is important to do the distinction as the value are not quite the same for select all

File: packages/types/src/inventory/common/inventory-level.ts [hunk 6]
```diff
@@ -53,7 +54,8 @@ export interface InventoryLevelDTO {
   deleted_at: string | Date | null
 }
 
-export interface FilterableInventoryLevelProps {
+export interface FilterableInventoryLevelProps
+  extends BaseFilterable<FilterableInventoryLevelProps> {
```
Comment (adrien2p) [inline packages/types/src/inventory/common/inventory-level.ts:59]: Q: Should we allow OperatorMap on the props? Also, why do we use NumericalComparisonOperator which is using another type of operator and why don't we rely on OperatorMap instead for those specific operator. It seems weird to have two differet type of possible operators for our modules e.g lt vs $lt

File: integration-tests/modules/__tests__/inventory/index.spec.ts (hunk 3 above)
Comment (olivermrbl) [reply integration-tests/modules/__tests__/inventory/index.spec.ts]: I believe I pointed that out above. Is this something else? 🤔

Comment (adrien2p) [reply integration-tests/modules/__tests__/inventory/index.spec.ts]: Unfortunately, your comment was far below and therefore i put this one before reaching yours

File: integration-tests/modules/__tests__/inventory/index.spec.ts (hunk 4 above)
Comment (pKorsholm) [reply integration-tests/modules/__tests__/inventory/index.spec.ts:387]: the item can definitely exist, just without being stocked at the location, i.e. no inventory level with that combination of properties exist. I can run another query just to validate the inventory-items passed along or change the message a bit? 

I think partly this is just due to the id I chose 😅

Comment (adrien2p) [reply integration-tests/modules/__tests__/inventory/index.spec.ts:387]: but is this id existing for the item?

**Comment by olivermrbl** [comment]
@adrien2p, OK for you to merge this?

**Pull request merged**